submissions/src/main/kotlin/gov/cdc/prime/reportstream/submissions/controllers/SubmissionController.kt

```diff
@@ -0,0 +1,185 @@
+
+            // Insert into Table
+            // TableEntity() sets PartitionKey and RowKey. Both are required by azure and combine to create the PK
+            val tableEntity = TableEntity(reportId.toString(), reportId.toString())
+            val tableProperties = mapOf(
+                "report_received_time" to reportReceivedTime.toString(),
+                "report_accepted_time" to reportReceivedTime.toString(), // Will be updated when the report is accepted
+                "report_id" to reportId.toString(),
+                "status" to status
+            )
+            tableClient.createEntity(tableEntity.setProperties(tableProperties))
+            logger.info("Inserted report into table storage: reportId=$reportId")
+
+            // Create and publish custom event
+            val reportReceivedEvent = ReportReceivedEvent(
+                timeStamp = reportReceivedTime,
+                reportId = reportId,
+                parentReportId = reportId,
+                rootReportId = reportId,
+                headers = filterHeaders(headers),
+                sender = clientId!!,
+                senderIP = headers["x-azure-clientip"].toString(),
+                fileSize = headers["content-length"].toString(),
+                blobUrl = blobClient.blobUrl
+            )
+            logger.debug("Created ReportReceivedEvent")
+
+            // Log to Application Insights
+            telemetryClient.trackEvent(
+                "ReportReceivedEvent",
+                mapOf("event" to objectMapper.writeValueAsString(reportReceivedEvent)),
+                null)
+            telemetryClient.flush()
+            logger.info("Tracked ReportReceivedEvent with Application Insights")
+
+            val response =
+                CreationResponse(
+                    reportId,
+                    status,
+                    OffsetDateTime.now(),
+                )
+            logger.info("Report submission successful: reportId=$reportId")
+
+            return ResponseEntity(response, HttpStatus.CREATED)
+        } catch (e: Exception) {
+            logger.error("Error during report submission: ${e.message}", e)
+            return ResponseEntity("Internal Server Error: ${e.message}", HttpStatus.INTERNAL_SERVER_ERROR)
+        }
+    }
+
+    private fun filterHeaders(headers: Map<String, String>): Map<String, String> {
+        val headersToInclude = listOf("client_id", "Content-Type", "payloadName")
+        return headers.filter { it.key in headersToInclude }
+    }
+
+    private fun formBlobName(
+        reportId: UUID,
+        headers: Map<String, String>,
+    ): String {
+        val senderName = headers["client_id"]?.lowercase()
+        return when (headers["Content-Type"]?.lowercase()) {
+            "application/hl7-v2" -> "receive/$senderName/$reportId.hl7"
+            "application/fhir+ndjson" -> "receive/$senderName/$reportId.fhir"
+            else -> "receive/$senderName/$reportId"
```

Comment:
I think we should throw?  Though based on the `consumes` annotation I guess this is  not possible?

Reply:
yeah I like throwing here too. It shouldn't ever happen due to `consumes` attribute but seems clearer this way